maintenance/wikia/email.php

```diff
@@ -0,0 +1,184 @@
+ini_set( 'error_reporting', E_NOTICE );
+
+// Eliminate the need to set this on the command line
+if ( !getenv( 'SERVER_ID' ) ) {
+	putenv( "SERVER_ID=177" );
+}
+
+require_once( dirname( __FILE__ ) . '/../Maintenance.php' );
+
+/**
+ * Class Email\CLI
+ */
+class EmailCLI extends Maintenance {
+
+	protected $verbose = false;
+	protected $test = false;
+	protected $to = '';
+	protected $type = '';
+	protected $argString = '';
+	protected $list = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Pre-populate LVS suggestions";
+		$this->addOption( 'test', 'Test mode; make no changes',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG,
+			't'
+		);
+		$this->addOption( 'verbose', 'Show extra debugging output',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG,
+			'v'
+		);
+
+		$this->addOption( 'to', 'Address to send email to',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'type', 'Type of email to send',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'args', 'Arguments for email type',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'list', 'List email types',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG
+		);
+	}
+
+	public function execute() {
+		try {
+			$this->initOptions();
+
+			if ( $this->list ) {
+				$this->showValidEmailTypes();
+				exit( 0 );
+			}
+
+			// The controller class should end with 'Controller'
+			$controllerClass = 'Email\\Controller\\' . $this->type . 'Controller';
+			$this->assertControllerExists( $controllerClass );
+
+			$params = $this->argStringToArray( $this->argString );
+			$resp = F::app()->sendRequest( $controllerClass, 'handle', $params, true );
+
+			$this->assertSuccessfulEmail( $resp );
+
+			echo "Email sent as '$this->to' successfully!\n";
+			exit( 0 );
+		} catch ( Exception $e ) {
+			echo "An error occurred:\n";
+			echo $e->getMessage() . "\n";
+			exit( 1 );
+		}
+	}
+
+	protected function initOptions() {
+		$this->test = $this->hasOption( 'test' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->to = $this->getOption( 'to', '' );
+		$this->type = $this->getOption( 'type', '' );
+		$this->argString = $this->getOption( 'args', '' );
+		$this->list = $this->getOption( 'list', false );
+
+		$this->assertRequiredOptionsExist();
+
+		// The list functionality doesn't need an email address
+		if ( !$this->list ) {
+			$this->setupUser();
+		}
+	}
+
+	protected function assertRequiredOptionsExist() {
+		if ( $this->list ) {
+			return;
+		}
+
+		if ( !empty( $this->to ) && !empty( $this->type ) ) {
+			return;
+		}
+
+		throw new OptionException( 'Either --list or both --to and --type options are required' );
+	}
+
+	protected function setupUser() {
+		// We either have a user ID or a user name
+		if ( preg_match( '/^\d+$/', $this->to ) ) {
```

Comment:
Looks useful, I'll use ```filter_var``` instead.  As far as the edge case, this will see pretty limited usage so I don't expect that to be a problem, but its easy enough to fix.